module/zfs/dnode.c

```diff
@@ -1625,6 +1638,16 @@ dnode_rele_and_unlock(dnode_t *dn, void *tag, boolean_t evicting)
 	}
 }
 
+/*
+ * Test whether we can create a dnode at the specified location.
+ */
+int
+dnode_try_claim(objset_t *os, uint64_t object, int slots)
+{
+	return (dnode_hold_impl(os, object, DNODE_MUST_BE_FREE|DNODE_DRY_RUN,
+	    slots, FTAG, NULL));
```

Comment:
nit: I'd suggest passing `NULL` instead of `FTAG` since it will not be used.  Additionally, why don't we add the following to the top of `dnode_hold_impl()` to make the interface absolutely clear.

```
IMPLY(flag & DNODE_DRY_RUN, (tag == NULL) && (dnp == NULL));
```